plugins/Goals/API.php

```diff
@@ -331,11 +332,29 @@ protected function getItems($recordName, $idSite, $period, $date, $abandonedCart
         $archive   = Archive::build($idSite, $period, $date, $segment);
         $dataTable = $archive->getDataTable($recordNameFinal);
 
-        $this->enrichItemsTableWithViewMetrics($dataTable, $recordName, $idSite, $period, $date, $segment);
+        // Before Matomo 4.0.0 ecommerce views were tracked in custom variables
+        // So if Matomo was installed before try to fetch the views from custom variables and enrich the report
+        if (version_compare(DbHelper::getInstallVersion(),'4.0.0-b2', '<')) {
+            $this->enrichItemsTableWithViewMetrics($dataTable, $recordName, $idSite, $period, $date, $segment);
```

Comment:
In the `enrichItemsTableWithViewMetrics` might need to check if the custom variables plugin is active since we will move it to the marketplace plus people might deactivate it?

Reply:
Might not be fully understanding it... are we there basically merging the metrics from custom variables with the metrics from log_link_visit_action if Matomo was installed before 4.0?

I suppose this would usually be mostly for data on the day they upgrade to Matomo 4 cause all previous day will have data from custom variables and all other ones from log_link_visit_action.

Reply:
That's basically for BC. If someone installs Matomo after 4 the variables will be directly tracked into the `log_link_visit_action` table, so there is no need to fallback to the old behavior.
If Matomo was installed before it might be possible that there were some ecommerce views tracked as custom variables. If those old data would be archived there wouldn't be any ecommerce views reported, so we try to get those data from the custom variables reports and merge that in. (That was the way it was done before).
Actually we should see in the UI tests that this behaviour works, as I didn't update the OmniFixture.sql, so it still includes ecommerce views as custom variables but is archived with the new logic before the tests run...